lib/src/controls/pickers/calendar_view.dart

```diff
@@ -0,0 +1,1084 @@
+    setState(() {
+      _scrollDate = null;
+      _anchorMonth = DateTime(date.year, date.month, 1);
+      _displayMode = CalendarViewDisplayMode.year;
+    });
+    if (_yearScrollController.hasClients) {
+      _yearScrollController.jumpTo(0);
+    }
+  }
+
+  /// Steps to the next or previous year based on the offset.
+  ///
+  /// If the offset is positive, it navigates to a future year. If negative,
+  /// it navigates to a past year.
+  void stepYear({int offset = 0}) {
+    navigateToYear(_yearForPage(offset));
+  }
+
+  bool _isInRange(DateTime day) {
+    if (_selectedStart == null || _selectedEnd == null) return false;
+    return !day.isBefore(_selectedStart!) && !day.isAfter(_selectedEnd!);
+  }
+
+  void _onDayTapped(DateTime day, bool inScope) {
+    if (widget.selectionMode == CalendarViewSelectionMode.none) return;
+    if (!inScope && !widget.isOutOfScopeEnabled) return;
+
+    setState(() {
+      if (widget.selectionMode == CalendarViewSelectionMode.single) {
+        _selectedStart = DateUtils.isSameDay(_selectedStart, day) ? null : day;
+        _selectedEnd = null;
+        widget.onSelectionChanged?.call(
+          CalendarSelectionData(
+            selectedDates: _selectedStart != null ? [_selectedStart!] : [],
+            startDate: _selectedStart,
+            endDate: null,
+          ),
+        );
+      } else if (widget.selectionMode == CalendarViewSelectionMode.range) {
+        if (_selectedStart == null || _selectedEnd != null) {
+          _selectedStart = day;
+          _selectedEnd = null;
+        } else if (_selectedStart != null && _selectedEnd == null) {
+          if (day.isBefore(_selectedStart!)) {
+            _selectedEnd = _selectedStart;
+            _selectedStart = day;
+          } else {
+            _selectedEnd = day;
+          }
+          widget.onSelectionChanged?.call(
+            CalendarSelectionData(
+              selectedDates: [?_selectedStart, ?_selectedEnd],
+              startDate: _selectedStart,
+              endDate: _selectedEnd,
+            ),
+          );
+        }
+      } else if (widget.selectionMode == CalendarViewSelectionMode.multiple) {
+        final alreadySelected = _selectedMultiple.any(
+          (d) => DateUtils.isSameDay(d, day),
+        );
+        if (alreadySelected) {
+          _selectedMultiple.removeWhere((d) => DateUtils.isSameDay(d, day));
+        } else {
+          _selectedMultiple.add(day);
+        }
+        _selectedMultiple.sort((a, b) => a.compareTo(b));
+        widget.onSelectionChanged?.call(
+          CalendarSelectionData(
+            selectedDates: List.unmodifiable(_selectedMultiple),
+            startDate: _selectedMultiple.isNotEmpty
+                ? _selectedMultiple.first
+                : null,
+            endDate: _selectedMultiple.length > 1
+                ? _selectedMultiple.last
+                : null,
+          ),
+        );
+      }
+    });
+  }
+
+  int get firstDayOfWeek {
```

Comment:
Documentation missing.
```suggestion
  /// Gets the first day of the week based on the current locale.
  int get firstDayOfWeek {
```